pkg/util/funcs/memoize_arg_noerror.go

```diff
@@ -0,0 +1,33 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package funcs
+
+import "sync"
+
+type memoizedArgNoErrorFunc[K comparable, T any] struct {
+	sync.Mutex
+	fn      func(K) T
+	results map[K]T
```

Comment:
Should we be concern with the map growing in size?

Reply:
Of that is true, do we need the `memoizedArgNoErrorFunc` for just holding a single entry? mostly asking to see if this is a bit of an early optimization

Reply:
In order to maintain the same function as before, all the users need to be using the same `*http.Client`. I added this, so that it properly varied based on the path provided, which it was not doing before.